packages/ingest/src/lock.ts

```diff
@@ -81,54 +86,56 @@ async function addLock(
 export async function removeLock(
   bucket: string,
   providerName: string,
-  filename: string
+  granuleId: string
 ): Promise<void> {
   await deleteS3Object(
     bucket,
-    `${lockPrefix}/${providerName}/${filename}`
+    `${lockPrefix}/${providerName}/${granuleId}`
   );
 }
 
 /**
  *
- * @param {string} bucket
- * @param {Object} provider
- * @param {string} provider.id
- * @param {number} provider.globalConnectionLimit
- * @param {string} filename
- * @param {number} counter
+ * @param {string} bucket - system bucket to place the lock files
+ * @param {object} provider - provider object
+ * @param {string} provider.id - provider id
+ * @param {number} provider.globalConnectionLimit - provider globalConnectionLimit
+ * @param {number} provider.maxDownloadTime - provider maxDownloadTime for a granule
+ * @param {string} granuleId - id of downloading granule
+ * @param {number} counter - retry counter
  * @returns {Promise<boolean>}
  */
 export async function proceed(
   bucket: string,
   provider: {
     id: string,
-    globalConnectionLimit?: number
+    globalConnectionLimit?: number,
+    maxDownloadTime?: number,
   },
-  filename: string,
+  granuleId: string,
   counter = 0
 ): Promise<boolean> {
-  if (provider.globalConnectionLimit === undefined) {
+  const { globalConnectionLimit, maxDownloadTime } = provider;
```

Comment:
I think this line should go after this check
```
if (globalConnectionLimit === undefined) {
    return true;
  }
```

Reply:
Actually, I am not sure anymore but am wondering if no globalConnectionLimit is provided, then below at 
```
if (count >= globalConnectionLimit) {
    log.debug(`The "${provider.id}" provider's globalConnectionLimit of "${provider.globalConnectionLimit}" has been reached.`);
    // wait for 5 second and try again
    await sleep(5000);
    return proceed(bucket, provider, granuleId, counter + 1);
  }
  ```

will it always be false since anything compared to `undefined` is false? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Greater_than#comparing_boolean_null_undefined_nan

Reply:
Ahh that makes sense